code/modules/antagonists/heretic/items/unfathomable_curio.dm

```diff
@@ -44,9 +44,29 @@
 	if(!(slot & slot_flags))
 		return
 
+	RegisterSignal(user, COMSIG_LIVING_CHECK_BLOCK, PROC_REF(shield_reaction))
+
 	if(!IS_HERETIC(user))
 		to_chat(user, span_warning("The curio wraps around you, and you feel the beating of something dark inside it..."))
 
+/obj/item/storage/belt/unfathomable_curio/dropped(mob/user)
+	. = ..()
+	UnregisterSignal(user, COMSIG_LIVING_CHECK_BLOCK)
+
+// Here we make sure our curio is only able to block while worn on the belt slot
+/obj/item/storage/belt/unfathomable_curio/proc/shield_reaction(mob/living/carbon/human/owner,
+	atom/movable/hitby,
+	damage = 0,
+	attack_text = "the attack",
+	attack_type = MELEE_ATTACK,
+	armour_penetration = 0,
+	damage_type = BRUTE,
+)
+	SIGNAL_HANDLER
+
+	if(SEND_SIGNAL(src, COMSIG_ITEM_HIT_REACT, owner, hitby, attack_text, 0, damage, attack_type) & COMPONENT_HIT_REACTION_BLOCK && (owner.belt == src))
```

Comment:
I can see we're already doing this in at least one other place, but we should restrict signals to one source and not just fire them off like proc calls.
If the base proc `hit_reaction` isn't appropriate here, then extract this part out into a reusable proc so we can track the source of this signal more cleanly.

Reply:
Do you mean like a generic proc that sends the signal? Sorta like how health scanners do it?

Reply:
why are we reusing this signal at all? isn't this is what hit_reaction is like, FOR?
@Ghommie you have blame for the first case of this, what is going on?

Reply:
I don't have any idea what my fault is in this. I can agree having the signals should generalize it (and perhaps have a linter test that wards us against multiple instances of the same signals being sent in the code)